django-oracle-drcp/base.py

```diff
@@ -20,17 +21,35 @@ def __init__(self, *args, **kwargs):
             raise ImproperlyConfigured('POOL database option requires \'min\', \'max\', and \'increment\'')
         if not all(isinstance(val, int) for val in pool_config.values()):
             raise ImproperlyConfigured('POOL database option values must be numeric')
+
+        options = self.settings_dict.get('OPTIONS', None)
+        threaded = options.get('threaded', False) if options else False
+
         self.pool = cx_Oracle.SessionPool(
             user=self.settings_dict['USER'],
             password=self.settings_dict['PASSWORD'],
-            dsn=self.settings_dict['NAME'], **pool_config)
+            dsn=self.get_dsn(),
+            threaded=threaded,
+            **pool_config)
 
     def get_new_connection(self, conn_params):
         conn_params.update({
             'pool': self.pool,
         })
         return super(DatabaseWrapper, self).get_new_connection(conn_params)
 
+    def get_dsn(self):
+        settings_dict = self.settings_dict
+
+        if settings_dict['PORT']:
+            dsn = self.Database.makedsn(settings_dict['HOST'],
+                                   int(settings_dict['PORT']),
+                                   settings_dict['NAME'])
+        else:
+            dsn = settings_dict['NAME']
+
+        return dsn
+
```

Comment:
@jmarente, the username and password are definitely needed in my environment.  In any case, if there is a problem with the base class _connect_string, then we should override it, and maybe omit USER and PASSWORD if they are not needed.

Can you point to a place in the OCI Programmer's Guide where they are not needed.